core/src/main/java/com/graphhopper/GraphHopper.java

```diff
@@ -979,8 +979,16 @@ protected void postProcessing(boolean closeEarly) {
 
         if (chPreparationHandler.isEnabled())
             chPreparationHandler.createPreparations(ghStorage);
-        if (!isCHPrepared())
+        if (isCHPrepared()) {
+            // check loaded profiles
+            for (CHProfile profile : chPreparationHandler.getCHProfiles()) {
+                if (!getProfileHash(profile.getProfile()).isEmpty()
```

Comment:
Why is `isEmpty()` treated special here? Why is it possible for the hash to be empty?

Reply:
Ah, I just blindly assumed the symmetry between CH and LM. So, yes. It is not necessary for CH as we want to load it here.